express-error-handling/server/index.js

```diff
@@ -0,0 +1,208 @@
+require('dotenv/config');
+const pg = require('pg');
+const express = require('express');
+const ClientError = require('./client-error'); // eslint-disable-line
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.get('/api/grades', (req, res, next) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.json(grades);
+    })
+    .catch(err => {
+      next(err);
+      // res.status(500).json({
+      //   error: 'an unexpected error occurred'
+      // });
+      throw new ClientError(500, 'an unexpected error occurred');
+    });
```

Comment:
In the `catch` you just need to call `next`, you don't want to also `throw` a `ClientError`
```JAVASCRIPT
.catch(err => {
  next(err);
});
```